sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/SyntaxLocation.cs

```diff
@@ -0,0 +1,46 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using ULocation = SonarAnalyzer.Protobuf.Ucfg.Location;
+
+namespace SonarAnalyzer.Security.Ucfg
+{
+    public static class SyntaxLocation
+    {
+        public static ULocation Get(Location location)
+        {
+            var lineSpan = location.GetLineSpan();
+            return new ULocation
+            {
+                FileId = location.SourceTree.FilePath,
```

Comment:
It would be weird if the location points to something that has no source tree. This does not run in SLVS so partially broken code should not be possible.